versioned_docs/version-4.4/building-extensions/index.md

```diff
@@ -2,10 +2,51 @@
 sidebar_position: 5
 ---
 Build Extensions
-=======================
-This is the content for tutorials how to build extensions
-:::caution TODO
+================
+Joomla is a rich featured content management system, but if you're building a website with Joomla and you need extra features which aren't available by default, then you can easily extend it with extensions. There are five common types of extensions for Joomla: Components, Modules, Plugins, Templates, and Languages. There are three others: Packages, Files and Libraries. Each of these extensions handle specific functionality (many built-in features of Joomla are implemented using extensions).
 
-This page is unfinished, please use the **Edit this Page** link at the bottom of this page to help make it more useful.
+![Screenshot showing extension types](./_assets/screenshot-extension-types.jpg)
 
-:::
+# Components
+[Components](./components/index.md) provide the central part of a web page on a Joomla site; each site web page displays the output from one component. They can be thought of as mini applications. Most components have two parts: a site part and an administrator part. For example, `com_content` is the component which handles articles; on the site front-end `com_content` displays articles to website visitors and on the back-end `com_content` provides the functionality for administrators to edit articles. 
+
+- Examples: Content (com_content), Banners (com_banners), Contacts (com_contact), News Feeds (com_newsfeeds)
+- Management feature: Admin menu → Components → Contacts (for com_contact)
+
+# Modules
+[Modules](./modules/index.md) are more lightweight and flexible extensions used within page rendering. These modules are mostly known as the “boxes” that are arranged around a component, for example: the login module or the breadcrumbs module. Modules are assigned per menu item. So you can decide to show or hide the login module depending on which menu item the user is viewing. Sometimes modules are linked to components such as the “Latest News” module which links to the `com_content` and displays links to the newest content items. However, modules do not need to be linked to components, as a matter of fact they don't even need to be linked to anything and can be just static HTML or text.
```

Comment:
>  Sometimes modules are linked to components ...

In general the module is a companion of the component. 
The role is to display a short teaser of component data (latest news), or specific data that come from component (like menu, image slider).

But also true, that module can be standalone, to display some weather info, or stuff like that.